test/snapp_auction.js

```diff
@@ -520,4 +521,31 @@ contract("SnappAuction", async (accounts) => {
       assert.equal(currentAuction.size, 2)
     })
   })
+
+  describe("Larger Test Cases", () => {
+    it("Fill order batch", async () => {
+      const instance = await SnappAuction.new()
+      await setupEnvironment(MintableERC20, instance, token_owner, [user_1], 2)
+      const order = encodeOrder(0, 1, 2, 3)
+      const maxAuctionSize = (await instance.AUCTION_BATCH_SIZE.call()).toNumber()
+      const orders = Array(maxAuctionSize).fill(order)
+      const partitionedOrders = splitArray(orders, 100)
+      let concatenated_orders
+      // let count = 0
+      await partitionedOrders.forEach(function(partition) {  // should be about 10
+        concatenated_orders = Buffer.concat(partition)
+        instance.placeSellOrders(concatenated_orders, { from: user_1 })
```

Comment:
Don't you need to return an awaitable here for the top level await to take an effect?

Reply:
Apparently not.

Reply:
Can you explain what we are awaiting on in the current version? As far as I can tell forEach returns `void` and await only does something when it is called on a Promise.

I think we have to await on `instance.placeSellOrders(concatenated_orders, { from: user_1 })` inside the forEach function (and declare that one as async).

You can if your current version is working by simulating a long transaction time 
```js
const waitFor = (ms) => new Promise(r => setTimeout(r, ms));
await partitionedOrders.forEach(async function(partition) {  // should be about 10
        concatenated_orders = Buffer.concat(partition)
        await waitFor(100)
        instance.placeSellOrders(concatenated_orders, { from: user_1 })
}
```

and check if your test still passes.